kratos/solving_strategies/builder_and_solvers/explicit_builder.h

```diff
@@ -0,0 +1,846 @@
+                    const SizeType eq_id = r_node_dofs[i_dof]->EquationId();
+                    const double aux_mass = elem_mass_vector(i_node * n_dofs + i_dof);
+// NOTE THAT THIS IS A BOTTLENECK--> WE ASSUME THAT WE WILL NOT USE THE SPACES IN HERE
+// #pragma omp critical
+//                     {
+//                     const double mass_value = TDenseSpace::GetValue(*mpLumpedMassVector, eq_id);
+//                     TDenseSpace::SetValue(*mpLumpedMassVector, eq_id, aux_mass + mass_value);
+//                     }
+#pragma omp atomic
+                    (*mpLumpedMassVector)[eq_id] += aux_mass;
+                }
+            }
+        }
+
+        // Set the lumped mass vector flag as true
+        mLumpedMassVectorIsInitialized = true;
+
+        KRATOS_CATCH("");
+    }
+
+    /**
+     * @brief Initialize the DOF set reactions
+     * For an already initialized dof set (mDofSet), this method sets to
+     * zero the corresponding reaction variable values. Note that in the
+     * explicit build the reactions are used as residual container.
+     */
+    virtual void InitializeDofSetReactions()
+    {
+        // Firstly check that the DOF set is initialized
+        KRATOS_ERROR_IF_NOT(mDofSetIsInitialized) << "Trying to initialize the explicit residual but the DOFs set is not initialized yet." << std::endl;
+        KRATOS_ERROR_IF(mEquationSystemSize == 0) << "Trying to set the equation ids. in an empty DOF set (equation system size is 0)." << std::endl;
+
+        // Loop the reactions to initialize them to zero
+        IndexPartition<int>(mEquationSystemSize).for_each(
+            [&](int i_dof){
+                auto it_dof = mDofSet.begin() + i_dof;
+                auto& r_reaction_value = it_dof->GetSolutionStepReactionValue();
+                r_reaction_value = 0.0;
+            }
+        );
+    }
+
+    /**
+     * @brief It computes the reactions of the system
+     * @param rModelPart The model part to compute
+     */
+    virtual void CalculateReactions()
+    {
+        if (mCalculateReactionsFlag) {
+            // Firstly check that the DOF set is initialized
+            KRATOS_ERROR_IF_NOT(mDofSetIsInitialized) << "Trying to initialize the explicit residual but the DOFs set is not initialized yet." << std::endl;
+            KRATOS_ERROR_IF(mEquationSystemSize == 0) << "Trying to set the equation ids. in an empty DOF set (equation system size is 0)." << std::endl;
+
+            // Calculate the reactions as minus the current value
+            // Note that we take advantage of the fact that Kratos always works with a residual based formulation
+            // This means that the reactions are minus the residual. As we use the reaction as residual container
+            // during the explicit resolution of the problem, the calculate reactions is as easy as switching the sign
+            IndexPartition<int>(mEquationSystemSize).for_each(
```

Comment:
better use a Blockpartition here, you don't need the index
=> I think you updated this in the strategy already

Reply:
Yes. I missed this one (and the one below).